model/events/LtiAgsListener.php

```diff
@@ -120,26 +135,53 @@ private function onDeliveryExecutionFinish(DeliveryExecutionState $event): void
                 }
             }
 
-            /** @var QueueDispatcherInterface $taskQueue */
-            $taskQueue = $this->getServiceLocator()->get(QueueDispatcherInterface::SERVICE_ID);
-            $taskQueue->createTask(new SendAgsScoreTask(), [
-                'retryMax' => $this->getAgsMaxRetries(),
-                'registrationId' => $user->getRegistrationId(),
-                'deliveryExecutionId' => $deliveryExecution->getIdentifier(),
-                'agsClaim' => $agsClaim->normalize(),
-                'data' => [
-                    'userId' => $user->getIdentifier(),
-                    'activityProgress' => ScoreInterface::ACTIVITY_PROGRESS_STATUS_COMPLETED,
-                    'gradingProgress' => $this->isManualScored($session)
-                        ? ScoreInterface::GRADING_PROGRESS_STATUS_PENDING_MANUAL
-                        : ScoreInterface::GRADING_PROGRESS_STATUS_FULLY_GRADED,
-                    'scoreGiven' => $scoreTotal,
-                    'scoreMaximum' => $scoreTotalMax,
-                ]
-            ], 'AGS score send on test finish');
+            $this->queueSendAgsScoreTaskWithScores(
+                'AGS score send on test finish',
+                $user->getLaunchData(),
+                $deliveryExecution,
+                $scoreTotal,
+                $scoreTotalMax,
+                $this->isManualScored($session)
+                    ? ScoreInterface::GRADING_PROGRESS_STATUS_PENDING_MANUAL
+                    : ScoreInterface::GRADING_PROGRESS_STATUS_FULLY_GRADED
+            );
         }
     }
 
+    private function queueSendAgsScoreTaskWithScores(
+        string $taskLabel,
+        LtiLaunchData $ltiLaunchData,
+        DeliveryExecutionInterface $deliveryExecution,
+        $scoreTotal,
+        $scoreTotalMax,
```

Comment:
missing type for $scoreTotal and $scoreTotalMax

Reply:
I specially removed it cause not yet sure which type it might be except float

Reply:
Ok but I will be prone to errors and strange hard to debug app behavior in future.